src/Eloquent/Model.php

```diff
@@ -270,6 +299,24 @@ public function fromJson($value, $asObject = false)
         return Json::decode($value ?? '', ! $asObject);
     }
 
+    /** @inheritdoc */
+    protected function castAttribute($key, $value)
+    {
+        $castType = $this->getCastType($key);
+
+        switch ($castType) {
+            case 'immutable_custom_datetime':
+            case 'immutable_datetime':
+                if (str_starts_with($this->getCasts()[$key], 'immutable_date:')) {
```

Comment:
I don't quite understand how the cast works here: why can you have `immutable_date:` and `immutable_datetime` at the same time?

Reply:
If you cast a data attribute like `immutable_date`, it's totally okay and works fine, but if you cast it to `immutable_date` with custom formatting like `immutable_date:j.n.Y H:i`, It will be returned as an `immutable_custom_datetime` from `getCastType` method. It's because Laravel doesn't handle the `immutable_custom_date` cast type. I might open a PR later on Laravel but until that time, it will be fixed in this way.